### PR TITLE
Removed Variable dtype

### PR DESCRIPTION
**Pull request by gsheni**
- Variable dtype is unused so we are removing the functionality.

### CODE REVIEW

File: featuretools/variable_types/variable.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def __eq__(self, other):
             self.entity_id == other.entity_id
 
     def __repr__(self):
-        ret = u"<Variable: {} (dtype = {})>".format(self.name, self.dtype)
+        ret = u"<Variable: {} >".format(self.name)
```
Comment (kmax12) [inline featuretools/variable_types/variable.py]: still print the dtype, just use `self._dtype_repr`

File: featuretools/entityset/entity.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def __repr__(self):
         repr_out = u"Entity: {}\n".format(self.id)
         repr_out += u"  Variables:"
         for v in self.variables:
-            repr_out += u"\n    {} (dtype: {})".format(v.id, v.dtype)
+            repr_out += u"\n    {} ".format(v.id)
```
Comment (kmax12) [inline featuretools/entityset/entity.py]: still print the dtype, just use `self._dtype_repr`

File: featuretools/variable_types/variable.py [hunk 3]
```diff
@@ -70,10 +70,10 @@ def create_from(cls, variable):
     def name(self):
         return self._name if self._name is not None else self.id
 
-    @property
-    def dtype(self):
-        return self._dtype_repr \
-            if self._dtype_repr is not None else "generic_type"
+    # @property
```
Comment (kmax12) [inline featuretools/variable_types/variable.py]: delete this

Comment (gsheni) [reply featuretools/variable_types/variable.py]: Fixed.

File: featuretools/entityset/entity.py (hunk 2 above)
Comment (gsheni) [reply featuretools/entityset/entity.py]: Fixed.

File: featuretools/variable_types/variable.py (hunk 1 above)
Comment (gsheni) [reply featuretools/variable_types/variable.py]: Fixed.

**Pull request merged**